code/game/verbs/ooc.dm

```diff
@@ -107,6 +107,181 @@
+			to_chat(src, span_warning("OOC is globally muted"))
+			return
+		if(prefs.muted & MUTE_OOC)
+			to_chat(src, span_warning("You cannot use OOC (muted)."))
+			return
+		if(handle_spam_prevention(msg, MUTE_OOC))
+			return
+		if(findtext(msg, "byond://"))
+			to_chat(src, span_danger("Advertising other servers is not allowed."))
+			log_admin_private("[key_name(usr)] has attempted to advertise in OOC: [msg]")
+			message_admins("[ADMIN_TPMONTY(usr)] has attempted to advertise in OOC: [msg]")
+			return
+
+	if(is_banned_from(ckey, "OOC"))
+		to_chat(src, span_warning("You have been banned from OOC."))
+		return
+
+	mob.log_talk(msg, LOG_OOC)
+
+	for(var/client/C in GLOB.clients)
+		if(!(C.prefs.toggles_chat & CHAT_OOC))
+			continue
+		if(!(C.mob in GLOB.xeno_mob_list) && !(C.mob in GLOB.observer_list) || check_other_rights(C, R_ADMIN, FALSE))
+			continue
+
+		var/display_name = mob.name
+		if(admin && !(mob in GLOB.xeno_mob_list))
+			display_name = mob.key
+
+		// Admins open straight to player panel
+		if(check_other_rights(C, R_ADMIN, FALSE))
+			display_name = "<a class='hidelink' href='?_src_=holder;[HrefToken(TRUE)];playerpanel=[REF(usr)]'>[display_name]</a>"
+		var/avoid_highlight = C == src
+		to_chat(C, "<font color='#665544'>[span_ooc("<span class='prefix'>XOOC: [display_name]")]: <span class='message linkify'>[msg]</span></span></font>", avoid_highlighting = avoid_highlight)
+
+	for(var/client/C in GLOB.admins)
+		if(!check_other_rights(C, R_ADMIN, FALSE) && (C.mob in GLOB.xeno_mob_list) && (C.mob in GLOB.observer_list))
+			continue
+		var/display_name = mob.name
+		if(admin && !(mob in GLOB.xeno_mob_list))
+			display_name = mob.key
+		if(check_other_rights(C, R_ADMIN, FALSE))
+			display_name = "<a class='hidelink' href='?_src_=holder;[HrefToken(TRUE)];playerpanel=[REF(usr)]'>[display_name]</a>"
+		var/avoid_highlight = C == src
+		to_chat(C, "<font color='#665544'>[span_ooc("<span class='prefix'>XOOC: [display_name]")]: <span class='message linkify'>[msg]</span></span></font>", avoid_highlighting = avoid_highlight)
+
+
+/client/verb/mooc_wrapper()
+	set hidden = TRUE
+	var/message = input("", "MOOC \"text\"") as null|text
+	ooc(message)
+
+
+/client/verb/mooc(msg as text) // Same as XOOC, but for humans.
+	set name = "MOOC"
+	set category = "OOC"
+
+	var/admin = check_rights(R_ADMIN, FALSE)
+
+	if(!mob)
+		return
+	if(IsGuestKey(key))
+		to_chat(src, "Guests may not use MOOC.")
+		return
+	if(mob.stat == DEAD && !admin)
+		to_chat(src, span_warning("You must be alive to use MOOC."))
+		return
+	if(!(mob in GLOB.human_mob_list) && !admin)
+		to_chat(src, span_warning("You must be a human to use MOOC."))
+		return
+
+	msg = copytext_char(sanitize(msg), 1, MAX_MESSAGE_LEN)
+
+	if(!msg)
+		return
+	if(NON_ASCII_CHECK(msg))
+		return
+
+	msg = emoji_parse(msg)
+
+	if(!(prefs.toggles_chat & CHAT_OOC))
+		to_chat(src, span_warning("You have OOC muted."))
+		return
+
+	if(!check_rights(R_ADMIN, FALSE))
+		if(!GLOB.ooc_allowed)
```

Comment:
a way to selectively mute each faction OOC would be nice IMO. Don't remove that though